lib/tasks/projects.rake

```diff
@@ -135,4 +135,19 @@ namespace :projects do
     number_to_sync = args.number_to_sync || 2000
     Project.least_recently_updated_stats.where(platform: supported_platforms).limit(number_to_sync).each{|project| project.update_maintenance_stats_async(priority: :low)}
   end
+
+  desc 'Set license_normalized flag'
+  task set_license_normalized: :environment do
+    supported_platforms = ["Maven", "NPM", "Pypi", "Rubygems", "Nuget", "Packagist"]
+    Project.where(platform: supported_platforms, license_normalized: false).find_in_batches do |group|
+      group.each do |project|
+        project.normalize_licenses
+        # check if we set a new value
+        if project.license_normalized_changed?
+          # update directly to skip any callbacks
```

Comment:
the comment makes me wonder what the callbacks might do and whether it's important

Reply:
The callbacks will be doing a lot, but nothing related to this value so I don't think it is worth running them. On save the Project will update it's Repository which also has callbacks, update the source rank, set up dependencies for the project, call all these methods on Project:
```
normalize_licenses
set_latest_release_published_at
set_latest_release_number
set_latest_stable_release_info
set_runtime_dependencies_count
set_language
```